### ISSUE TITLE
No RSSI value when pressing remote button

### ISSUE THREAD

**Issue by bradphillips921** [rstrouse/ESPSomfy-RTS#14]
I followed the hardwire guide and setup guide exactly. I have the same ESP32 board and transceiver from the hardware guide and have it wired per the guide.  I have the firmware installed and the ESP32 connected to my network. I'm able to access the GUI, I've set my RX bandwidth and Frequency deviation per the setup guide. But when I press the button on my Somfy remote I don't get any value for the RSSI. I've tried multiple Somfy channels, checked my wiring, etc. But still can't get an RSSI number when pressing the button on the remote while right next to the transceiver.

**Comment by rstrouse** [comment]
Which CC1101 transceiver and ESP32 are you using?

**Comment by FlaMike** [comment]
I have the same challenge (onto the next one??).  Hardware setup is exactly like yours, Dupont cabling until setup is complete.  ESP is a WROOM-32 Devkit1 Version 1 and the transceiver is a E07-M1101D-SMA with the duck antenna, it's connected to the board, of course.

I tried both 50 and 80 bits for the radio and the radio is saved and enabled.

I am running ESP Somfy RTS version 1.4.5.  The initial config I am using is exactly what's in  the Wiki for software configuation: 56 bit radio, RX bandwidth 96.96, Frequency deviation 11.43, TX power 10 db.  Pins set as in the Wiki.

**Comment by rstrouse** [comment]
In your setup change TX to GPIO12 and RX to GPIO13.  These appear to be documented backwards.  I added the ETH boards and it has different pin assignments.  I think I got lost in the markdown.

**Comment by FlaMike** [comment]
Will do.  Thanks!

**Comment by rstrouse** [comment]
Just change it in the pin assignments page do not move the pins around.

**Comment by FlaMike** [comment]
👍

**Comment by FlaMike** [comment]
I changed the pin assignments on the Somfy Settings page, saved the radio, and  rebooted the device from the General tab.  The "Radio Not Initialized" banner is not disappearing when I save the radio.  I tried a hard reboot as well, but the radio does not appear to be initializing.  Possible dead board from the far east?  I have another I can try...

**Comment by rstrouse** [comment]
Double and triple check your wiring.  The pins numbers are from the top of the board not the bottom.  The way that thing is marked is a menace.  Pin 1 is the left bottom pin with the pins facing away from you,

**Comment by rstrouse** [comment]
> I apologize for being a major irrigation to your backside. I have no excuse for my failure to follow directions.

No worries sometimes I stick my backside out the window at the carwash when I need a good irrigation.

**Comment by bradphillips921** [comment]
Changing those TX and RX pins didn't work for me. I have TX set at GPIO-12 and RX set at GPIO-13. I also tried the opposite with TX set to GPIO-13 and RX set to GPIO-12. Neither worked.

GPIO-13 is connected to terminal 3 on the transmitter. GPIO-12 is connected to terminal 8 on the transmitter

I'm using this transmitter
https://www.amazon.com/dp/B09CTRYV4L?psc=1&ref=ppx_yo2ov_dt_b_product_details

And this ESP board
https://www.amazon.com/dp/B09XDMVS9N?psc=1&ref=ppx_yo2ov_dt_b_product_details

It's wired per the instructions on the hardware page.
https://github.com/rstrouse/ESPSomfy-RTS/wiki/Simple-ESPSomfy-RTS-device

**Comment by rstrouse** [comment]
That looks like a fine bill of materials for this project.  Did the transceiver come with an antenna if it did not you need one?  

When you are connecting the transceiver the pin numbers are from the top where the pins are facing away from you.  Please make verify your connections with this in mind.

**Comment by bradphillips921** [comment]
Good catch! I don't have the male pin in the antenna. I just ordered one. I'll report back if that worked.

**Comment by rstrouse** [comment]
I thought that might be an RP-SMA antenna.  Make sure you get one tuned to 433Mhz.

**Comment by bradphillips921** [comment]
I bought this antenna. Unfortunately, it's still not working.
https://www.amazon.com/dp/B072Q7319V?psc=1&ref=ppx_yo2ov_dt_b_product_details

Here are my settings
https://drive.google.com/file/d/1xGhHDzgG24nKCQGnnWmuu4ZhpVwab_20/view?usp=sharing

**Comment by Anashost** [comment]
> I bought this antenna. Unfortunately, it's still not working. https://www.amazon.com/dp/B072Q7319V?psc=1&ref=ppx_yo2ov_dt_b_product_details
> 
> Here are my settings https://drive.google.com/file/d/1xGhHDzgG24nKCQGnnWmuu4ZhpVwab_20/view?usp=sharing

my esp32 boards has dedicated rx and tx pins other than GPIO-12 and GPIO-13.  so if your board have them try that. in the espsomfy settengs, these pins usually translate to TX: GPIO-01 and RX: GPIO-03.

**Comment by rstrouse** [comment]
You can use the pins suggested by Anashost if you like but you should simply change your setup to TX = GPIO-13 and RX = GPIO-12.  I need to go in and edit the docs to reflect the pictures.  The pins can be switched so long as GDO0 is connected to the TX and GDO2 is connected to the RX pin.

**Comment by bradphillips921** [comment]
It's still not working for me. I've confirmed that the transceiver pin 3 is connected to GD00-TX (labeled D13 on the board) and transceiver pin 8 is connected to GD02-RX (labeled D12 on the board). I have TX = GPIO-13 and RX = GPIO-12 on the configuration screen.
https://drive.google.com/open?id=1xGhHDzgG24nKCQGnnWmuu4ZhpVwab_20&authuser=bphillips%40phillipsstafford.com&usp=drive_fs

But I'm still not getting an RSSI signal when I push the remote button within 6 inches of the transceiver antenna.

**Comment by rstrouse** [comment]
Change the pins to the following and try again.

![image](https://user-images.githubusercontent.com/47839015/230701288-1327d0f2-9df3-44c2-99ba-b47ee83b0111.png)

**Comment by Anashost** [comment]
This is the diagram of his esp32

![Screenshot_20230408_060355_Chrome](https://user-images.githubusercontent.com/41167157/230702627-4b2ccf59-6f60-40eb-b38d-e2cbf312651b.jpg)

I think 12 & 13 doesn't work as TX & RX in his case (they didn't work for my board either).
So maybe he should use the pins that supports that, so:
01 & 03 (the ones i used)
Or
16 & 17

I've an ESP32-WROOM-32D with SILAPS chip , and i had to use pins 01 & 03 as tx rx. I'm using the blue transceiver with the spring antenna. these are my configs:

![Screenshot_20230408_025257_Chrome](https://user-images.githubusercontent.com/41167157/230702671-d57fa50e-075c-4cfb-9951-fd99293bcece.jpg)

I might be wrong here as I'm new to this world, and still learning. Hope this helps.

**Comment by rstrouse** [comment]
I too am new to this world and the SiLabs folks don't make it easy with all the versions of the ESP32 and very little information on what the differences are.  Near as I can figure all the current WROOM models being shipped are using the ESP32-D0WD chip and all of them are dual-core.  This means that an ESP32S and an ESP32D are exactly the same.  At one point the S was supposed to mean single core and D was supposed to be dual code.  To make matters worse, they also say DEVKIT01 on them.

I listened to so many YouTube videos from the guy with the Swiss accent that I even started to pick up his accent but was still confused by the models.   From a layperson perspective, I would say the board pictured above is the same one I have.

The software does not use a serial UART so it never initializes any of the pins to the UART.  Using pins 1 and 3 are supposedly the same as using 12 and 13 or any of the other un-reserved pins.

Here is Andreas going through them.
https://www.youtube.com/watch?v=LY-1DHTxRAk

**Comment by bradphillips921** [comment]
I tried the recommendation from Anashost but that didn't work.

I set my pin numbers how you showed in your latest post, rstrouse. Still doesn't work.

**Comment by rstrouse** [comment]
@bradphillips921 I ordered the antenna that you purchased above and it works fine with several different transceivers that I tested.  I also made an attempt to review the wiring from your pictures and it looks correct.   I am starting to think you have a bad transceiver.  I have bought several of these (simply because they show up in a day) and only had one that was bad and acted like yours is acting.

https://www.amazon.com/dp/B01DS1WUEQ?psc=1&ref=ppx_yo2ov_dt_b_product_details

**Comment by bradphillips921** [comment]
Yep, I'm leaning towards that as well. I just ordered transceivers from the link you posted. I should have them next week. I'll test with those.

**Comment by rstrouse** [comment]
@bradphillips921 keep that antenna.  I did a bit of tuning on one of my devices and it increased the receive strength by about 30% even when it was already spectacular.

**Comment by reindruu** [comment]
It looks like I have the same setup and the same problem.
Not picking up any buttons from Somfy remotes.
I bought them at Ali (maybe that is the problem?!?):
ESP 32 board 30 Pin CP2102 chip:
[https://www.aliexpress.com/item/1005001627605230.html?spm=a2g0o.order_detail.order_detail_item.8.31a96d76agcUp8]
Transmitter incl. antenna:
[https://www.aliexpress.com/item/33059343309.html?spm=a2g0o.order_detail.order_detail_item.12.31a96d76agcUp8]

Configuration is ok, Radio is initialized, header disappeared.



![SomfySettings](https://user-images.githubusercontent.com/111060508/231248574-7e6af3fb-ead4-4aba-939c-36b9174d93b1.jpg)
![IMG-8764](https://user-images.githubusercontent.com/111060508/231248603-d18fcfca-70ce-4a3a-b626-690bbeca08f3.jpg)
![SomfyFirmware](https://user-images.githubusercontent.com/111060508/231249048-76648577-9a55-4113-be93-c03847d99369.jpg)

**Comment by Anashost** [comment]
I just want to mention that i got mine from Aliexpress as well, but everything works great 👍🏻

ESP32-WROOM-32D - SILAPS CHIP
https://a.aliexpress.com/_EQlnZHl

CC1101 (Blue with spring antenna)
https://a.aliexpress.com/_EJV3HKB

You may got a faulty board, or it's something else!

**Comment by Anashost** [comment]
@reindruu If you're going to order the same boards as mine, then here is the pins I've used.
![Screenshot_20230408_010530_AliExpress](https://user-images.githubusercontent.com/41167157/231267203-b8f3db85-04ef-473e-8cc9-fef57a069cae.jpg)

Configurations:
![Screenshot_20230410_213500_Chrome](https://user-images.githubusercontent.com/41167157/231267592-f80644d9-2bcf-4044-a2f5-71ce56012055.jpg)

**Comment by rstrouse** [comment]
Are those RTS remotes or IO remotes?  If it is IO, they are not 433Mhz protocol RTS.

**Comment by reindruu** [comment]
> 
@rstrouse 
Oops................ 
You are right, completely.
![IMG-8766](https://user-images.githubusercontent.com/111060508/231275244-4c8fc8eb-40d4-4c51-96da-aa654e9ac53a.jpg)

**Comment by rstrouse** [comment]
I'm so sorry.  Unfortunately, this solution will not work for IO motors and your journey to control your own blinds will not end here.  What a terrible trend all this cloud based IoT stuff is.

**Comment by reindruu** [comment]
> I'm so sorry. Unfortunately, this solution will not work for IO motors and your journey to control your own blinds will not end here. What a terrible trend all this cloud based IoT stuff is.

Big disappointment. I did not know there were different motors. Life's full of disappointments but is was an experience.
Thanks for your huge good manual and guidlines.
Maybe your next project is the IO motor................

**Comment by rstrouse** [comment]
@bradphillips921 if your new hardware does not solve the problem please open another issue.  This one got a bit hard to follow and thanks for the antenna tip.
